src/hotspot/os/linux/os_linux.cpp

```diff
@@ -4120,13 +4129,17 @@ char* os::Linux::reserve_memory_special_huge_tlbfs(size_t bytes,
                                                    size_t alignment,
                                                    char* req_addr,
                                                    bool exec) {
+  // Select large_page_size from _page_sizes
+  // that is smaller than size_t bytes
+  size_t large_page_size = os::page_size_for_region_aligned(bytes, 1);
```

Comment:
This is partly what I'm looking at from a slightly different direction. And my current thinking is that we should get rid of all mappings that are not properly aligned when using large pages. But that is something for a different PR.

I need to look more at this next week, but for this to work as before this call needs to use the unaligned version: `os::page_size_for_region_unaligned(...)` otherwise we will here get a small page size in many cases and that can not be used with the code doing the reservations below.